crates/biome_js_analyze/src/lint/nursery/no_useless_length_check.rs

```diff
@@ -0,0 +1,366 @@
+};
+use biome_console::markup;
+use biome_js_syntax::{
+    AnyJsExpression, AnyJsLiteralExpression, AnyJsMemberExpression, JsBinaryExpression,
+    JsBinaryOperator, JsLogicalExpression, JsParenthesizedExpression, JsSyntaxKind, T,
+};
+use biome_rowan::{AstNode, BatchMutationExt, TextRange};
+
+use crate::JsRuleAction;
+
+declare_lint_rule! {
+    /// Disallow unnecessary length checks within logical expressions.
+    ///
+    /// - `Array#some()` returns `false` for an empty array. There is no need to check if the array is not empty.
+    /// - `Array#every()` returns `true` for an empty array. There is no need to check if the array is empty.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// if (array.length === 0 || array.every(Boolean));
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// if (array.length !== 0 && array.some(Boolean));
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// if (array.length > 0 && array.some(Boolean));
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// const isAllTrulyOrEmpty = array.length === 0 || array.every(Boolean);
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// if (array.every(Boolean));
+    /// ```
+    ///
+    /// ```js
+    /// if (array.some(Boolean));
+    /// ```
+    ///
+    /// ```js
+    /// const isAllTrulyOrEmpty = array.every(Boolean);
+    /// ```
+    ///
+    /// ```js
+    /// const isAllTrulyOrEmpty = array.every(Boolean);
+    /// ```
+    ///
+    /// ```js
+    /// if (array.length === 0 || anotherCheck() || array.every(Boolean));
+    /// ```
+    ///
+    /// ```js
+    /// const isNonEmptyAllTrulyArray = array.length > 0 && array.every(Boolean);
+    /// ```
+    ///
+    /// ```js
+    /// const isEmptyArrayOrAllTruly = array.length === 0 || array.some(Boolean);
+    /// ```
+    ///
+    pub NoUselessLengthCheck {
+        version: "next",
+        name: "noUselessLengthCheck",
+        language: "js",
+        recommended: false,
+        sources: &[RuleSource::EslintUnicorn("no-useless-length-check")],
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+#[derive(Clone, Debug)]
+pub enum ErrorType {
+    UselessLengthCheckWithSome,
+    UselessLengthCheckWithEvery,
+}
+
+/// Whether the node is a descendant of a logical expression.
+fn is_logical_exp_descendant(node: &AnyJsExpression, operator: JsSyntaxKind) -> bool {
+    let Some(parent) = node.syntax().parent() else {
+        return false;
+    };
+    parent
+        .ancestors()
+        .find_map(|ancestor| {
+            if let Some(logical_exp) = JsLogicalExpression::cast(ancestor.clone()) {
+                return logical_exp
+                    .operator_token()
+                    .ok()
+                    .map(|token| token.kind() == operator)
+                    .or(Some(false));
+            }
+            (!JsParenthesizedExpression::can_cast(ancestor.kind())).then_some(false)
```

Comment:
`JsParenthesizedExpression` can be tough to deal with; for example,  I don't think we are able to catch these cases:

```js
if (((((array.length === 0 || array.every(Boolean))))));
```

The type `AnyJsExpression` has a method called `omit_parentheses`, which returns the expression inside all the parenthesis. You should use that, and see if you can catch cases like the one I shared

Reply:
This function returns whether the parent is a `JsLogicalExpression`. When the parent is a `JsParenthesizedExpression`, we want to continue searching up to its parent as well, so this function is called to check for that. Therefore, even with the current implementation, such cases can be detected. (However, since I have learned about the convenient `omit_parentheses` function, I would like to rewrite some parts using it. Thank you.)